cmd/kubelet/app/server.go

```diff
@@ -603,6 +604,17 @@ func getReservedCPUs(machineInfo *cadvisorapi.MachineInfo, cpus string) (cpuset.
 }
 
 func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Dependencies, featureGate featuregate.FeatureGate) (err error) {
+	var healthChecker watchdog.HealthChecker
+	if utilfeature.DefaultFeatureGate.Enabled(features.SystemdWatchdog) {
+		// NewHealthChecker returns an error indicating that the watchdog is configured but the configuration is incorrect,
+		// the kubelet will not be started.
+		healthChecker, err = watchdog.NewHealthChecker()
+		if err != nil {
+			return fmt.Errorf("create health checker: %w", err)
+		}
+		healthChecker.Start(ctx)
```

Comment:
Do we need a `Start` to be a separate method? Should we just start it from constructor?

Reply:
> Do we need a `Start` to be a separate method? Should we just start it from constructor?

Agree with you. There's no need to separate Start in the current scenario.